src/helpers/content/preprocess-external-content.ts

```diff
@@ -0,0 +1,88 @@
+import MarkdownIt from "markdown-it";
+import { JSDOM } from "jsdom";
+import OpenAI from "openai";
+
+export class ExternalContentProcessor {
+  private readonly _md = new MarkdownIt();
+
+  constructor(
+    private readonly _openAi: OpenAI,
+    private readonly _llmModel: string
+  ) {}
+
+  public async evaluateExternalElements(comment: GithubCommentScore) {
+    const html = this._md.render(comment.content.replaceAll("\r", "\n"));
+    const jsDom = new JSDOM(html);
+
+    if (jsDom.window.document.body) {
+      const htmlElement = jsDom.window.document.body;
+      await this._handleAnchorElements(htmlElement);
+      await this._handleImageElements(htmlElement);
+    } else {
+      throw new Error(`Could not create DOM for comment [${JSON.stringify(comment)}]`);
+    }
+  }
+
+  private async _handleAnchorElements(htmlElement: HTMLElement) {
+    const anchors = htmlElement.getElementsByTagName("a");
+    for (const anchor of anchors) {
+      const href = anchor.getAttribute("href");
+      if (!href) continue;
+      const linkResponse = await fetch(href);
+      const contentType = linkResponse.headers.get("content-type");
+      if (!contentType || !contentType.startsWith("text/")) continue;
+      const linkContent = await linkResponse.text();
+      const llmResponse = await this._openAi.chat.completions.create({
+        model: this._llmModel,
+        messages: [
+          {
+            role: "system",
+            content: "You are a helpful assistant that evaluates external content and summerizes it.",
+          },
+          {
+            role: "user",
+            content: `Summarize the following external content in one paragraph, written in a single line: ${linkContent}`,
+          },
+        ],
+      });
+      const altContent = llmResponse.choices[0]?.message?.content;
+      if (!altContent) continue;
+      // Should replace the content in the comment
+      console.log(altContent);
+    }
+  }
+
+  private async _handleImageElements(htmlElement: HTMLElement) {
+    const images = htmlElement.getElementsByTagName("img");
+    for (const image of images) {
+      const href = image.getAttribute("href");
+      if (!href) continue;
+      const linkResponse = await fetch(href);
+      const contentType = linkResponse.headers.get("content-type");
+      if (!contentType || !contentType.startsWith("image/")) continue;
+      const imageData = await linkResponse.arrayBuffer();
+      const base64Image = Buffer.from(imageData).toString("base64");
+      const llmResponse = await this._openAi.chat.completions.create({
+        model: this._llmModel,
+        messages: [
+          {
+            role: "system",
+            content: "You are a helpful assistant that evaluates external images and summerizes it.",
+          },
+          {
+            role: "user",
+            content: [
+              { type: "text", text: "Describe this image concisely in one paragraph, written in a single line." },
```

Comment:
I wonder if the new deepseek r1 model is best for this - because its reasoning capabilities is supposed to be almost the best of all the latest models, but it's also very cheap. 

I think that for summarization, reasoning could be a useful intermediary step. We should also strive to provide far more context in the prompt for better results with reasoning.